qf_lib/data_providers/data_provider.py

```diff
@@ -282,13 +282,11 @@ def _get_valid_latest_available_prices(start_date: datetime, tickers: Sequence[T
 
     @staticmethod
     def _compute_start_date(nr_of_bars_needed: int, end_date: datetime, frequency: Frequency):
-        if frequency <= Frequency.DAILY:
-            nr_of_days_to_go_back = math.ceil(nr_of_bars_needed * (365 / 252) + 10)
-        else:
-            days_per_year = Frequency.DAILY.occurrences_in_year
-            bars_per_year = frequency.occurrences_in_year
-            bars_per_day = int(bars_per_year / days_per_year)
-            nr_of_days_to_go_back = math.ceil(nr_of_bars_needed / bars_per_day + 1)
+        calendar_to_business_day_factor = 365 / 252
+        frequency_multiplier = Frequency.DAILY.value / frequency.value
+        margin = 10 if frequency <= Frequency.DAILY else 1
+        nr_of_days_to_go_back = math.ceil(
+            nr_of_bars_needed * frequency_multiplier * calendar_to_business_day_factor) + margin
```

Comment:
Actually as Frequency.DAILY.value == 252 this could be even more simplified to
```
nr_of_days_to_go_back = math.ceil(nr_of_bars_needed * 365 / frequency.value ) + margin
```

Reply:
correct